vaadin-grid-flow-integration-tests/frontend/benchmark.js

```diff
@@ -0,0 +1,186 @@
+import {
+  registerStyles,
+  css,
+} from '@vaadin/vaadin-themable-mixin/register-styles.js';
+
+var style = document.createElement('style');
+style.type = 'text/css';
+style.appendChild(
+  document.createTextNode(`
+  @keyframes content-ready {
+    from {
+      opacity: 1;
+    }
+    to {
+      opacity: 1;
+    }
+  }
+
+  vaadin-grid-tree-toggle[leaf] {
+    transition: opacity 1s;
+    opacity: 0.9;
+  }
+`)
+);
+document.head.appendChild(style);
+
+registerStyles(
+  'vaadin-grid',
+  css`
+    [part~='body-cell'] {
+      height: 50px;
+    }
+
+    @keyframes content-ready {
+      to {
+        opacity: 1;
+      }
+    }
+
+    [part~='cell'] {
+      animation: content-ready 1s;
+    }
+
+    :host {
+      transition: opacity 1s;
+    }
+
+    :host[loading] {
+      opacity: 0.9;
+    }
+  `
+);
+
+registerStyles(
+  'vaadin-*',
+  css`
+    @keyframes content-ready {
+      to {
+        opacity: 1;
+      }
+    }
+
+    :host {
+      animation: content-ready 1s;
+    }
+  `
+);
+
+const whenRendered = (grid) => {
+  return new Promise((resolve) => {
+    let readyTimer;
+    let endTime;
+    const listener = (e) => {
+      if (e.animationName === 'content-ready' || e.propertyName === 'opacity') {
+        endTime = performance.now();
+        readyTimer && clearTimeout(readyTimer);
+        if (!grid.loading) {
+          readyTimer = setTimeout(() => {
+            grid.$.scroller.removeEventListener('animationstart', listener);
+            grid.removeEventListener('animationstart', listener);
+            grid.removeEventListener('transitionstart', listener);
+            resolve(performance.now() - endTime);
```

Comment:
Unless I've missed something, this could be simplified by just resolving `endTime`.

Then `measureRender` below would be simplified from:
`reportResult(performance.now() - start - idleTime)`
to:
`reportResult(endTime - start)`

Reply:
Good catch, done.